GeneratorPgn.py

```diff
@@ -0,0 +1,318 @@
+                castling[0] = 1
+            elif figure == 'Q':
+                castling[1] = 1
+            elif figure == 'k':
+                castling[2] = 1
+            elif figure == 'q':
+                castling[3] = 1
+        en_passant = np.zeros((1, 8, 8,), dtype=np.float32)
+        if en_passant_str != '-':
+            en_passant[0, first_cords[en_passant_str[0]], second_cords[en_passant_str[1]]] = 1.
+        half_moves = int(half_moves_str)
+        move_idx = int(move_idx_str)
+        return {'board': board_position,
+                'castling': castling,
+                'en_passant': en_passant,
+ 
+                'player': next_mover,
+                'half_move': half_moves, 'full_move': move_idx,
+                'fen': raw_fen}
+ 
+    def preprocess_batch(self, batch):
+        """
+        Applies `preprocess_fen` for a batch of data.
+        :param batch: batch to preprocess
+        :return: dictionary of preprocessed tensors
+        """
+        boards = []
+        players = []
+        castlings = []
+        en_passants = []
+        q_values = []
+        half_moves = []
+        full_moves = []
+        fens = []
+        for fen in batch:
+            data = self.preprocess_fen(fen)
+            boards.append(data['board'])
+            players.append(data['player'])
+            castlings.append(data['castling'])
+            en_passants.append(data['en_passant'])
+            half_moves.append(data['half_move'])
+            full_moves.append(data['full_move'])
+            fens.append(data['fen'])
+ 
+        return {'boards': boards,
+                'players': players,
+                'castlings': castlings,
+                'en_passants': en_passants,
+                'half_moves': half_moves,
+                'full_moves': full_moves}
+ 
+    def __iter__(self):
+        """
+        Like `Iterator.__iter__()`, but applies preprocessing to yielded data.
+        :return: yields preprocessed batches of samples
+        """
+        for batch in self.iterator:
+            #borr=chess.Board().set_fen(batch)
+            #print(borr)
+            yield self.preprocess_batch(batch),batch
+ 
+
+castling=["K","Q","k","q"]
+
+def ArrayToFen(arr):
+    """
+    converts the data returned by the olo class to fen
+    :return: list of chessboard positions in FEN notation
+    """
+    fen=[]
+    for i in range(len(arr['players'])):
+        fen.append("")
+        pieces=["P","N","B","R","Q","K","p","n","b","r","q","k"]
+        players=["w","b"]
+        for line in range(8):
+            space=0
+            for field in range(8):
+                max_value=0
+                max_value_piece=""
+                for piece in range(12):
+                    if(arr['boards'][i][piece][field][line]>max_value):
+                        max_value = arr['boards'][i][piece][field][line]
+                        max_value_piece = pieces[piece]   
+                if(max_value>0.5):

```

Comment:
use black to automatically format this file - it lacks spaces around operators etc

besides, it contains redundant parentheses around the condition